docs/src/specification/encoding.rst

```diff
@@ -5,30 +5,42 @@ Most repo commands and status reports are Data packets whose Content contains
 ``RepoCommandParam`` or ``RepoCommandRes`` structure.
 These Data are issued via Pub-Sub protocol.
 Each ``RepoCommandParam`` and ``RepoCommandRes`` contains
-multiple ``ObjectParam`` and ``ObjectResult``, resp.
+multiple ``ObjParam`` and ``ObjStatus``, resp.
 These structures are defined as follows:
 
```

Comment:
Please define how to handle TLV-TYPE evolvability in this protocol.

Reply:
I don't know the answer but it seems not supported now.
@tianyuan129 Do we want to design a better protocol? The current version looks messy.

Reply:
> it seems not supported now

You could declare "all TLV-TYPE numbers are critical".
In this case, any future addition would be a breaking change.

Reply:
> I don't know the answer but it seems not supported now. @tianyuan129 Do we want to design a better protocol? The current version looks messy.

I kinda agree. The current Sync implementation is just quick code, time to revisit. But when you say a better protocol, are you referring to Sync itself or the entire repo command framework? The current repo command is very much object storage orientated, it could be hard to harmoniously put Sync function into the protocol.

Reply:
I mean the repo command framework. I think it needs a revisit to leave spaces for Sync and probably my Linked Object proposal.